homeassistant/components/number/__init__.py

```diff
@@ -133,6 +133,7 @@ class NumberEntityDescription(EntityDescription):
     native_step: float | None = None
     step: None = None
     unit_of_measurement: None = None  # Type override, use native_unit_of_measurement
+    mode: NumberMode | None = None
```

Comment:
All looks good to me, the only comment I have is: Can we keep this sorted alphabetically?

Reply:
Sure and done